hare/eligibility/oracle.go

```diff
@@ -76,20 +83,28 @@ func (o *Oracle) BuildVRFMessage(id types.NodeId, layer types.LayerID, round int
 	return w.Bytes(), nil
 }
 
-// IsEligible checks if id is eligible on the given layer where msg is the VRF message, sig is the role proof and assuming commSize as the expected committee size
-func (o *Oracle) IsEligible(commSize uint32, id types.NodeId, layer types.LayerID, msg, sig []byte) (bool, error) {
+func (o *Oracle) activeSetSize(layer types.LayerID) uint32 {
+	fmt.Println("the epoc is: ", safeLayer(layer).GetEpoch(o.layersPerEpoch))
+	return o.activeSetProvider.ActiveSetSize(safeLayer(layer).GetEpoch(o.layersPerEpoch) - 1)
+}
+
+// Eligible checks if id is eligible on the given layer where msg is the VRF message, sig is the role proof and assuming commSize as the expected committee size
+func (o *Oracle) Eligible(layer types.LayerID, round int32, committeeSize int, id types.NodeId, sig []byte) (bool, error) {
+	msg, err := o.buildVRFMessage(id, layer, round)
+
+	if err != nil {
+		log.Error("Could not build VRF message")
+		return false, err
+	}
+
 	// validate message
-	if res, err := o.vrf.Verify(msg, sig); !res {
+	if res, err := o.vrf(msg, sig, id.VRFPublicKey); !res {
```

Comment:
Also, you need to validate that `err` is nil..

Reply:
Why is it critical to check if it is nil (if it can produce no nil deref)?
In the case above it should be the responsability of o.vrf func to check it.

Reply:
You didn't respond to my first comment. `id.VRFPublicKey` is never assigned.

Reply:
As a rule of thumb - you always check for errors. always.
If `err!=nil` you shouldn't assume anything regarding the value of `res`. Here you only check the value of `res` which might be garbage if there was an error

Reply:
My response was in regards to id.VRFPublicKey and not the err nil (which is clear to me).

Reply:
You shouldn't be using the ID unless both fields are set.